<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>command_editor</class>
 <widget class="QDialog" name="command_editor">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>716</width>
    <height>521</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>716</width>
    <height>521</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>716</width>
    <height>521</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Comand Editor</string>
  </property>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>695</width>
     <height>497</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QCheckBox" name="cbConnect">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>&amp;Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QToolButton" name="tbExpand">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>24</width>
              <height>25</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Expand All&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>&amp;+</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="tbCollapse">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>24</width>
              <height>25</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Collapse All&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>&amp;-</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTreeWidget" name="twCommandTemplateTree">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Command</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Arguments</string>
           </property>
          </column>
          <item>
           <property name="text">
            <string>arduboy</string>
           </property>
           <property name="flags">
            <set>ItemIsDragEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
           <item>
            <property name="text">
             <string>clear</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </item>
           <item>
            <property name="text">
             <string>setCursor</string>
            </property>
            <property name="text">
             <string>%i %i &lt;x&gt; &lt;y&gt;</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>print</string>
            </property>
            <property name="text">
             <string>%s &lt;a text&gt;</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>display</string>
            </property>
           </item>
          </item>
          <item>
           <property name="text">
            <string>debug</string>
           </property>
           <property name="text">
            <string>%i 1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>debug</string>
           </property>
           <property name="text">
            <string>%i 0</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="gbCommands">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>401</width>
            <height>211</height>
           </size>
          </property>
          <property name="title">
           <string>Command (TX)</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <widget class="QPlainTextEdit" name="pteCommands">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>281</width>
             <height>181</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>281</width>
             <height>181</height>
            </size>
           </property>
           <property name="plainText">
            <string/>
           </property>
          </widget>
          <widget class="QPushButton" name="pbClearTx">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>300</x>
             <y>70</y>
             <width>99</width>
             <height>27</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>99</width>
             <height>27</height>
            </size>
           </property>
           <property name="text">
            <string>C&amp;lear</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pbSend">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>300</x>
             <y>30</y>
             <width>99</width>
             <height>27</height>
            </rect>
           </property>
           <property name="maximumSize">
            <size>
             <width>99</width>
             <height>27</height>
            </size>
           </property>
           <property name="text">
            <string>&amp;Send</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gbReception">
          <property name="minimumSize">
           <size>
            <width>401</width>
            <height>211</height>
           </size>
          </property>
          <property name="title">
           <string>Reception</string>
          </property>
          <widget class="QPlainTextEdit" name="pteReception">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>30</y>
             <width>281</width>
             <height>181</height>
            </rect>
           </property>
           <property name="maximumSize">
            <size>
             <width>281</width>
             <height>181</height>
            </size>
           </property>
           <property name="undoRedoEnabled">
            <bool>false</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="plainText">
            <string/>
           </property>
          </widget>
          <widget class="QPushButton" name="pbClearRx">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>300</x>
             <y>30</y>
             <width>99</width>
             <height>27</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>99</width>
             <height>27</height>
            </size>
           </property>
           <property name="text">
            <string>Cl&amp;ear</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <spacer name="horizontalSpacer">
            <property name="minimumSize">
             <size>
              <width>43</width>
              <height>23</height>
             </size>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pbClose">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>99</width>
              <height>27</height>
             </size>
            </property>
            <property name="text">
             <string>Cl&amp;ose</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pbClose</sender>
   <signal>clicked(bool)</signal>
   <receiver>command_editor</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>624</x>
     <y>482</y>
    </hint>
    <hint type="destinationlabel">
     <x>542</x>
     <y>445</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbClearRx</sender>
   <signal>clicked(bool)</signal>
   <receiver>pteReception</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>615</x>
     <y>301</y>
    </hint>
    <hint type="destinationlabel">
     <x>509</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbConnect</sender>
   <signal>clicked(bool)</signal>
   <receiver>gbCommands</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>322</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbConnect</sender>
   <signal>clicked(bool)</signal>
   <receiver>pteCommands</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>355</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbConnect</sender>
   <signal>clicked(bool)</signal>
   <receiver>pbClearTx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>621</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbConnect</sender>
   <signal>clicked(bool)</signal>
   <receiver>pbSend</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>19</y>
    </hint>
    <hint type="destinationlabel">
     <x>636</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tbCollapse</sender>
   <signal>clicked(bool)</signal>
   <receiver>twCommandTemplateTree</receiver>
   <slot>collapseAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>25</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>90</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tbExpand</sender>
   <signal>clicked(bool)</signal>
   <receiver>twCommandTemplateTree</receiver>
   <slot>expandAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>19</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>65</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbClearTx</sender>
   <signal>clicked(bool)</signal>
   <receiver>pteCommands</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>102</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbClearRx</sender>
   <signal>clicked(bool)</signal>
   <receiver>pteReception</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>645</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
